cmd/juju/romulus/sla/sla.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+// The sla package contains the implementation of the juju sla
+// command.
+package sla
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/romulus/api/sla"
+	"gopkg.in/macaroon.v1"
+
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/api/modelconfig"
+	"github.com/juju/juju/cmd/modelcmd"
+)
+
+// authorizationClient defines the interface of an api client that
+// the command uses to create an sla authorization macaroon.
+type authorizationClient interface {
+	// Authorize returns the sla authorization macaroon for the specified model,
+	Authorize(modelUUID, supportLevel, budget string) (*macaroon.Macaroon, error)
+}
+
+type slaClient interface {
+	SetSLALevel(level string, creds []byte) error
+	SLALevel() (string, error)
+}
+
+var newSlaClient = func(conn api.Connection) slaClient {
+	return modelconfig.NewClient(conn)
+}
+
+var newAuthorizationClient = func(options ...sla.ClientOption) (authorizationClient, error) {
+	return sla.NewClient(options...)
+}
+
+var modelId = func(conn api.Connection) string {
+	// Our connection is model based so ignore the returned bool.
+	tag, _ := conn.ModelTag()
+	return tag.Id()
+}
+
+// TODO (mattyw) See juju/cmd/juju/storage/show.go for a better way of doing this.
+// TODO (mattyw) This should be fixed before this lands in master.
+// NewSLACommand returns a new command that is used to set sla credentials for a
+// deployed application.
+func NewSLACommand() cmd.Command {
+	slaCommand := &supportCommand{
+		newSlaClient:           newSlaClient,
+		newAuthorizationClient: newAuthorizationClient,
+	}
+	slaCommand.newAPIRoot = slaCommand.NewAPIRoot
+	return modelcmd.Wrap(slaCommand)
+}
+
+// supportCommand is a command-line tool for setting
+// Model.SLACredential for development & demonstration purposes.
+type supportCommand struct {
```

Comment:
s/supportCommand/slaCommand/  otherwise it's confusing.. NewSLACommand returns a supportCommand instance..